### PR TITLE
Bug 1808358 - Use OnboardingPage component in NotificationPermissionDialogScreen

### PR DESCRIPTION
**Pull request by rahulsainani**
– Using `OnboardingPage` added in #28630 in NotificationPermissionDialogScreen.

### Next Steps
– Use `OnboardingPage` in `Onboarding` (rename to `UpgradeOnboarding`)

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### QA

- [x] **QA Needed**

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-debug` task.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation

Used by GitHub Actions.

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/onboarding/view/NotificationPermissionDialogScreen.kt [hunk 1]
```diff
@@ -83,145 +53,13 @@ fun NotificationPermissionDialogScreen(
             onDismiss()
             Onboarding.notifPppNegativeBtnClick.record(NoExtras())
         },
+        modifier = Modifier
+            .statusBarsPadding()
+            .navigationBarsPadding(),
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/onboarding/view/NotificationPermissionDialogScreen.kt:58]: Wondering if we should move this into `OnboardingPage`

**Review by gabrielluong — COMMENTED** [review]
Changes look good to me. Since we opted to land this after the Fenix migration, I didn't approve it to discourage folks from landing this last minute.

**Comment by gabrielluong** [comment]
Hey @rahulsainani, the team is in the process of moving this repository to [firefox-android](https://github.com/mozilla-mobile/firefox-android) early next week, and won’t be able to review or land any PRs until after the migration. Please feel free to reopen a new PR against the new repo once migration is complete. You will have to clone the new repository and reapply your changes. Sorry for the inconvenience.

**Comment by rahulsainani** [comment]
Thanks for the heads up @gabrielluong ! Makes sense, will open this on the monorepo 👍

**Pull request closed without merging**